.github/workflows/main.yml

```diff
@@ -61,17 +61,17 @@ jobs:
     # so, seperating steps with if conditions
     - name: build ROS2 for foxy
       if: ${{ matrix.ros_distro == 'foxy' }}
-      uses: ros-tooling/setup-ros@v0.6
+      uses: ros-tooling/setup-ros@0395475ab6416edc274bb3c43673b2091a217e02 #v0.6
```

Comment:
since I saw failing in the FOXY distro, I think this hash is the reasons.
I think it should be replaced with this hash:
236ab287884fd5a314fc030e91b2017abb46719e
See: https://github.com/ros-tooling/setup-ros/commits/v0.6/

Reply:
changed it to 236ab287884fd5a314fc030e91b2017abb46719e (v0.6 0.6.3)